### PR TITLE
fix(driver): fixed build against linux 6.6.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area driver-bpf
/area driver-modern-bpf

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Fixes build against linux 6.6, partly reverting https://github.com/falcosecurity/libs/pull/1349.
The kernel changes that https://github.com/falcosecurity/libs/pull/1349 addressed were dropped at the `rc3` kernel stage, and we do not need them anymore (https://www.phoronix.com/news/Linux-Revert-MG-Timestamps).
We are not at kernel rc6, therefore we expect the 6.6 release next week and no further breaking changes.
Basically, `I_CTIME_QUERIED` disappeared from kernel sources at `rc3`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(driver): fixed build against linux 6.6.
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Worfklow that attempts the build of this branch HEAD against 6.6: https://github.com/falcosecurity/libs/actions/runs/6531983900/job/17734288714

**Comment by FedeDP** [comment]
Head of branch tested: https://github.com/falcosecurity/libs/actions/runs/6532320274/job/17735265380
:green_circle:

**Review by LucaGuerra — APPROVED** [review]
